bin/faucet/src/main.rs

```diff
@@ -123,9 +123,11 @@ async fn main() -> anyhow::Result<()> {
                 )
                 .with_state(faucet_state);
 
-            let listener = TcpListener::bind((config.endpoint.host.as_str(), config.endpoint.port))
-                .await
-                .context("Failed to bind TCP listener")?;
+            let socket_addr = config.endpoint.socket_addrs(|| None)?.into_iter().next().ok_or(
+                anyhow::anyhow!("Couldn't get any socket addrs for endpoint: {}", config.endpoint),
+            )?;
```

Comment:
Would note this as a risk to fail if we don't e2e test this flow in CI before deploy

Reply:
Would it be enough to test that the default endpoints can be bound? This could be added to each of the config tests that you've created.